### PR TITLE
feature 01-crud-api

### PR DESCRIPTION
**Pull request by samasno**
## Trello Ticket
https://trello.com/c/LLaHNMXJ/15-preliminary-crud-api-for-description-service

## How to Setup
clone repo.
run mongodb service.
In project dir run:
`npm install`
`npm run seed`
`npm run server-dev`
use postman or a any client of the sort to send PUT and DELETE requests to:
`http://localhost:4004/description/:productId`(1-100)
and POST requests to:
`http://localhost:4004/description/new`
schema for POST and PUT requests in `database/database.js`
for test purposes if body is empty description will be automatically generated

verify record changes in db or view in browser at GET `http://localhost:4004/:productId`

for api tests run:
`npm run test`

## Purpose of Pull Request
did some minor reorganization of code and added endpoints for creating, updating, and deleting description records.

### CODE REVIEW

File: database/data.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ var getAvailableColors = (id, color) => {
 };
 
 //Generate data function
-var generateData = () => {
+var generateData = (start = 1, end = 100) => {
```
Comment (sjamcox) [inline database/data.js:121]: Smart update 👍

File: server/index.js [hunk 2]
```diff
@@ -29,43 +22,22 @@ app.get('*.js', function(req, res, next) {
...
-      let itemData = description;
-      res.status(200).send(itemData);
-      next();
-    }
-  });
-});
+app.route('/description/:productId')
+  .get(descriptionById)
+  .put(genUpdate, updateDescription)
+  .delete(deleteDescription);
```
Comment (sjamcox) [inline server/index.js:35]: This is CLEAN! I've never seen it done this way before. Your controllers file along with this really makes for excellent code clarity.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## Trello Ticket
+https://trello.com/###
+
+## How to Setup
+> Place instructions on how to run your pull request locally
+
+## Purpose of Pull Request
+> Place description of what you did and why
+
```
Comment (sjamcox) [inline .github/pull_request_template.md:9]: I'm gonna have to start using this. 🤔

**Review by sjamcox — APPROVED** [review]
Overall very clean. Ran your code and everything checks out. Interested to hear more about the middleware approach and how you came about it.

File: server/index.js (hunk 2 above)
Comment (samasno) [reply server/index.js:35]: express has some really nifty features like this for cleaning up routers.

**Pull request merged**